### PR TITLE
fix: multiple tags issue

### PR DESCRIPTION
**Pull request by ryuk156**
This PR fixes the issue of tags before ModuleSite can only use a single tag now it can use multiple tags.
It introduces a new system to handle tags length that is greater than the two so that card structure can be maintained

Before:
tags: "Asset"

Now:
tags: ["Asset","Library"]

Screenshot:
image

Note:  You will not find the above card that is in the image it was generated during testing

### CODE REVIEW

File: src/components/Cards/Cards.jsx [hunk 1]
```diff
@@ -4,15 +4,29 @@ import { Row, Col, Badge } from "reactstrap";
 import { GatsbyImage } from "gatsby-plugin-image";
 
 const Cards = ({ title, cover, tags, excerpt, path }) => {
+  let totalTags = [];
+  if (tags.length >= 1) {
+    for (let tag = 0; tag <= 1; tag++) {
+      totalTags.push(tags[tag]);
+    }
+  }
+  let tagCount = tags.length - 2;
```
Comment (pollend) [inline src/components/Cards/Cards.jsx:13]: why do you need to do this first. just use tags as is.

Comment (pollend) [inline src/components/Cards/Cards.jsx]: why do you need to do this first. just use tags as is. 

`{tags.map((tag) => <Badge className="mr-2">{tag}</Badge>}`
`{(tags.length -2) > 0 ? "+" : ""}`

**Review by pollend — CHANGES_REQUESTED** [review]
the lo

File: src/components/Cards/Cards.jsx (hunk 1 above)
Comment (ryuk156) [reply src/components/Cards/Cards.jsx]: The if statement check for tags length and  for loop inside it collect first two tags in array
On return we just display the array in which we collected first two's tags and then  , we also check for if our array is of length 2 or less, if it is less then we do not display anything, else if it is greater then 2, we subtract 2 from it and display + "value we get after subtract"

So basically ,
 Tag1 Tag2 +1 more

Conclusion:
We some how have to display first two tags 

The above code you suggested will display all the tags and the  "+ more " , we don't want all tags

Comment (ryuk156) [reply src/components/Cards/Cards.jsx]: Did some searching and got solution for removing this  if statement and loop , we can use aarry.slice(0,2) method in JS, I will  fix this tonight

File: src/components/Cards/Cards.jsx [hunk 2]
```diff
@@ -4,15 +4,23 @@ import { Row, Col, Badge } from "reactstrap";
...
           <div className="card-img search-cards">
             <GatsbyImage image={cover.gatsbyImageData} />
           </div>
-          <div className="md-tag mt-3 ml-3">
-            <Badge>{tags}</Badge>
+          <div className="d-flex">
+            <div className="md-tag mt-3 ml-3">
+              {tags.slice(0, 2).map((tag) => {
+                return <Badge className="mr-2">{tag}</Badge>;
+              })}
```
Comment (pollend) [inline src/components/Cards/Cards.jsx:19]: I believe you don't need the return if I remember my JSX correctly.

Comment (ryuk156) [reply src/components/Cards/Cards.jsx:19]: We can directly do this thing removing curly brackets and return , but we are passing html elements so returning them looks clear and readable,

**Comment by jdrueckert** [comment]
I wanted to test this, but I don't think we currently have any card with two tags, do we?
Can you introduce a second tag to one of the test cards for the sake of example, please?

**Comment by ryuk156** [comment]
> I wanted to test this, but I don't think we currently have any card with two tags, do we?
> Can you introduce a second tag to one of the test cards for the sake of example, please?

sure

**Review by jdrueckert — COMMENTED** [review]
When I click on the module cards, I get the following:
image

Is this expected / known?

**Comment by ryuk156** [comment]
> When I click on the module cards, I get the following:
> image
> 
> Is this expected / known?

I tested it few minutes ago it was working fine for me , let me check again

**Comment by jdrueckert** [comment]
Works for clicking on cards now, but now I get a similar error when clicking on one of the tags in the "opened" card:
image

**Review by pollend — APPROVED** [review]

File: src/components/Cards/Cards.jsx (hunk 2 above)
Comment (pollend) [reply src/components/Cards/Cards.jsx:19]:
```suggestion
              {tags.slice(0, 2).map((tag) => <Badge className="mr-2">{tag}</Badge>)}
```

File: src/components/PostTags/PostTags.jsx [hunk 3]
```diff
@@ -5,16 +5,19 @@ import { Button } from "reactstrap";
...
-    {tags && 
-        <Link
-          key={tags}
-          style={{ textDecoration: "none" }}
-          to={`/${type}/?keywords=&filter=${_.kebabCase(tags.split(" ")[0])}`}
-        >
-          <Button color="primary" size="lg">{tags}</Button>
-        </Link>
-      }
+    {console.log(tags)}
```
Comment (pollend) [inline src/components/PostTags/PostTags.jsx]: why do you need this logging here

**Review by pollend — CHANGES_REQUESTED** [review]

File: src/templates/modulelist.jsx [hunk 4]
```diff
@@ -44,7 +44,13 @@ const modulelist = (
...
           const searchRgx = new RegExp(escapeRegExp(searchQuery), "gi");
           const tagRgx = new RegExp(escapeRegExp(filterTag), "gi");
-          return module.tags.match(tagRgx) && module.title.match(searchRgx);
+          let matchedTag = [];
+          module.tags.map((tag) => {
+            if (tag != null) {
+              matchedTag.push(tag.match(tagRgx));
+            }
+          });
+          return matchedTag.toString().match(tagRgx) && module.title.match(searchRgx);
```
Comment (pollend) [inline src/templates/modulelist.jsx]:
```suggestion
          const matchedTag = blog.tags.filter(tag => tag != null).map(t => t.match(tagRgx))
          return matchedTag.toString().match(tagRgx) && blog.title.match(searchRgx);
```
I think you can do something like this. your using a map wrong with this expression. map is suppose to map A --> B. javascript does have a foreach prototype if I remember correctly.

File: src/templates/modulelist.jsx [hunk 5]
```diff
@@ -44,7 +44,13 @@ const modulelist = (
         DATA.filter((module) => {
```
Comment (pollend) [inline src/templates/modulelist.jsx]: the const DATA here is strange to me.

File: src/templates/blog.jsx [hunk 6]
```diff
@@ -47,7 +47,11 @@ const blog = (
         DATA.filter((blog) => {
           const searchRgx = new RegExp(escapeRegExp(searchQuery), "gi");
           const tagRgx = new RegExp(escapeRegExp(filterTag), "gi");
-          return blog.tags.match(tagRgx) && blog.title.match(searchRgx);
+          let matchedTag=[]
+          blog.tags.map(tag=>{
+            matchedTag.push(tag.match(tagRgx))
+          })
```
Comment (pollend) [inline src/templates/blog.jsx]:
```suggestion
         const matchedTag = blog.tags.map(t => t.match(tagRgx))
```

**Review by pollend — CHANGES_REQUESTED** [review]

File: src/templates/modulelist.jsx (hunk 5 above)
Comment (ryuk156) [reply src/templates/modulelist.jsx]: done

File: src/templates/blog.jsx (hunk 6 above)
Comment (ryuk156) [reply src/templates/blog.jsx]: done

File: src/templates/modulelist.jsx (hunk 4 above)
Comment (ryuk156) [reply src/templates/modulelist.jsx]: done, foreach will not return modified array

Comment (ryuk156) [reply src/templates/modulelist.jsx]: > done , foreach will not return modified array

**Review by jdrueckert — COMMENTED** [review]
Tests out fine, so :+1: from a UX perspective.
@pollend @majordwarf feel free to approve and merge once you think it's also good to go code-wise

**Review by pollend — APPROVED** [review]
this seems good

File: src/components/PostTags/PostTags.jsx (hunk 3 above)
Comment (pollend) [reply src/components/PostTags/PostTags.jsx]: can you remove this @ryuk156

**Review by pollend — CHANGES_REQUESTED** [review]
remove that console.log portion and I think this should be good anything else @majordwarf

**Comment by ryuk156** [comment]
> remove that console.log portion and I think this should be good anything else @majordwarf

done

**Review by pollend — APPROVED** [review]

**Pull request merged**